controllers/osbuild_controller.go

```diff
@@ -47,16 +82,267 @@ type OSBuildReconciler struct {
+		err = fmt.Errorf(errorMsg)
+		logger.Error(err, "postCompose request was failed")
+		r.updateOSBuildStatus(ctx, logger, osBuild, errorMsg, failedContainerBuild, emptyComposeID, emptyComposeID)
+		return err
+	}
+
+	containerComposeId := composerResponse.JSON201.Id.String()
+	logger.Info("postComposer request was sent and trigger a new compose ID %s", containerComposeId)
+
+	r.updateOSBuildStatus(ctx, logger, osBuild, edgeContainerJobStillRunningMsg, startedContainerBuild, containerComposeId, emptyComposeID)
+	return nil
+}
+
+func (r *OSBuildReconciler) updateOSBuildStatus(ctx context.Context, logger logr.Logger, osBuild *osbuildv1alpha1.OSBuild,
+	msg string, typeCondition osbuildv1alpha1.OSBuildConditionType, containerComposeId string, isoComposeId string) {
+	patch := client.MergeFrom(osBuild.DeepCopy())
+	if containerComposeId != emptyComposeID {
+		osBuild.Status.ContainerComposeId = containerComposeId
+	}
+
+	if isoComposeId != emptyComposeID {
+		osBuild.Status.IsoComposeId = isoComposeId
+	}
+
+	if osBuild.Status.Conditions == nil {
+		osBuild.Status.Conditions = []osbuildv1alpha1.OSBuildCondition{}
+	}
+
+	osBuild.Status.Conditions = append(osBuild.Status.Conditions, osbuildv1alpha1.OSBuildCondition{
+		Type:    typeCondition,
+		Message: &msg,
+	})
+
+	errPatch := r.OSBuildRepository.PatchStatus(ctx, osBuild, &patch)
+	if errPatch != nil {
+		logger.Error(errPatch, "Failed to patch OSBuild status")
+	}
+}
+
+func (r *OSBuildReconciler) checkComposeIDStatus(ctx context.Context, logger logr.Logger, composeID string) (composer.ComposeStatusValue, error) {
+	response, err := r.ComposerClient.GetComposeStatus(ctx, composeID)
+	if err != nil {
+		logger.Error(err, fmt.Sprintf("failed to get compose ID %s status", composeID))
+		return "", err
+	}
+	composerResponse, err := composer.ParseGetComposeStatusResponse(response)
+	if err != nil {
+		logger.Error(err, "failed to parse getCompose response")
+		return "", err
+	}
+	if composerResponse.JSON200 != nil {
+		return composerResponse.JSON200.Status, nil
+	}
+	return "", fmt.Errorf("something went wrong with requesting the composeID %v", composerResponse.StatusCode())
+}
+
+func (r *OSBuildReconciler) createImageRequest(targetImage osbuildv1alpha1.TargetImage, targetImageType osbuildv1alpha1.TargetImageType) composer.ImageRequest {
+	uploadOptions := composer.UploadOptions(composer.AWSS3UploadOptions{Region: ""})
+
+	imageRequest := composer.ImageRequest{
+		Architecture:  string(targetImage.Architecture),
+		ImageType:     composer.ImageTypes(targetImageType),
+		Ostree:        (*composer.OSTree)(targetImage.OSTree.DeepCopy()),
+		UploadOptions: &uploadOptions,
+	}
+	return imageRequest
+}
+
+func (r *OSBuildReconciler) createCustomizations(osbuildCustomizations *osbuildv1alpha1.Customizations) composer.Customizations {
+	if &osbuildCustomizations == nil {
```

Comment:
You should check `osbuildCustomizations == nil`, it's already the pointer you need to check